XenAdmin/Dialogs/EvacuateHostDialog.cs

```diff
@@ -45,906 +45,870 @@
-
-                // Move the panel containing the VM listbox up to fill the gap where the labels were
-                const int pad = 6;
-                int extraSize = panel2.Top - labelMasterBlurb.Top - pad;
-                panel2.Top = labelMasterBlurb.Top + pad;
-                panel2.Height += extraSize;
-            }
+            if (!_host.IsMaster() || connection.Cache.HostCount <= 1)
+                tableLayoutPanelNewMaster.Visible = false;
+            else
+                tableLayoutPanelPSr.Visible = false;
+
+            tableLayoutPanelWlb.Visible = _pool.IsVisible() && Helpers.WlbEnabled(_host.Connection) &&
+                                          WlbServerState.GetState(_pool) == WlbServerState.ServerState.Enabled;
 
-            Pool pool = Core.Helpers.GetPool(host.Connection);
-            if (Helpers.WlbEnabled(host.Connection) && WlbServerState.GetState(pool) == WlbServerState.ServerState.Enabled)
-                lableWLBEnabled.Visible = true;
+            tableLayoutPanelSpinner.Visible = false;
+            tableLayoutPanelStatus.Visible = false;
+            progressBar1.Visible = false;
 
-            vms = new Dictionary<string, VmPrecheckRow>();
-            this.host.PropertyChanged += new PropertyChangedEventHandler(hostUpdate);
+            _host.PropertyChanged += hostUpdate;
+            _pool.PropertyChanged += poolUpdate;
 
             ActiveControl = CloseButton;
         }
 
-        void NewMasterComboBox_DrawItem(object sender, DrawItemEventArgs e)
+        protected override void OnShown(EventArgs e)
+        {
+            base.OnShown(e);
+            Text = string.Format(Messages.EVACUATE_HOST_DIALOG_TITLE, _host.Name());
+            Rebuild();
+        }
+
+        private void Rebuild()
         {
-            if (NewMasterComboBox.Enabled)
+            if (_rebuildInProgress || hostAction != null && !hostAction.IsCompleted)
             {
-                using (SolidBrush backBrush = new SolidBrush(NewMasterComboBox.BackColor))
-                {
-                    e.Graphics.FillRectangle(backBrush, e.Bounds);
-                }
+                _rebuildRequired = true;
+                return;
+            }
 
-                if (e.Index == -1)
-                    return;
+            Program.Invoke(this, () =>
+            {
+                _rebuildInProgress = true;
+                tableLayoutPanelStatus.Visible = false;
+                progressBar1.Visible = false;
+                DisableButtons();
+                ClearVMs();
+                ClearHosts();
+                labelSpinner.Text = Messages.SCANNING_VMS;
+                spinnerIcon1.StartSpinning();
+                tableLayoutPanelSpinner.Visible = true;
+                Scan();
+            });
+        }
+
+        private void Scan()
+        {
+            //Save Evacuated VMs for later
+            var saveVMsAction = new DelegatedAsyncAction(connection,

```

Comment:
I suggest that we add suppressHistory=true to this action (in the same way we already have for the scanning action below), because it is confusing to see a "Saving VM properties" event on opening the Host evacuation dialog